### ISSUE TITLE
There may be a memory leak

### ISSUE THREAD

**Issue by 123shang60** [rust-lang/regex#742]
#### What version of regex are you using?

1.4.3

#### Describe the bug at a high level.

When used in large quantities, the memory will continue to grow

#### What are the steps to reproduce the behavior?

```
use regex::bytes::Regex;

fn main() {
    loop {
        let input = "NGS>";
        let reg = Regex::new(r#"(?m)(\x1b|\[[0-9;]*[a-zA-Z])"#).unwrap();
        let _ = reg.replace_all(input.as_bytes(), "".as_bytes());
    }
}
```

#### What is the actual behavior?

After running for 10 minutes, the memory consumption increased to 1GB.

**Comment by BurntSushi** [comment]
Does this still happen if you disable all features except for `std`?

**Comment by 123shang60** [comment]
yes

**Comment by Marwes** [comment]
Seems to be an older regression. So far I have bisected this to have been introduced in 1.3.3 . https://github.com/rust-lang/regex/compare/1.3.2...1.3.3 only shows a dependency update on thread_local

**Comment by Marwes** [comment]
Digging deeper this seems to be a regression in thread_local 1.1.1

**Comment by BurntSushi** [comment]
I suspected as much, which is why I asked if the leak occurred with all features except for `std` disabled. That would disable the use of `thread_local`.

I believe `thread_local 1.1.1` had some big changes internally. cc @Amanieu

**Comment by Marwes** [comment]
Made a fix https://github.com/Amanieu/thread_local-rs/pull/30

**Comment by Marwes** [comment]
Now released in 1.1.3 so this could be closed
